include/framework/runtime.hpp

```diff
@@ -0,0 +1,80 @@
+#ifndef FRAMEWORK_EVEREST_RUNTIME_HPP
+#define FRAMEWORK_EVEREST_RUNTIME_HPP
+
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <everest/logging.hpp>
+#include <fmt/color.h>
+#include <fmt/core.h>
+#include <framework/ModuleAdapter.hpp>
+#include <sys/prctl.h>
+
+namespace Everest {
+
+namespace po = boost::program_options;
+namespace fs = boost::filesystem;
+
+// FIXME (aw): should be everest wide or defined in liblog
+const int DUMP_INDENT = 4;
+
+// FIXME (aw): we should also define all other config keys and default
+//             values here as string literals
+const std::string MAIN_BIN_PATH = "bin/main";
+
+const auto TERMINAL_STYLE_ERROR = fmt::emphasis::bold | fg(fmt::terminal_color::red);
+const auto TERMINAL_STYLE_OK = fmt::emphasis::bold | fg(fmt::terminal_color::green);
+
+struct RuntimeSettings {
+    fs::path main_dir;
+    fs::path main_binary;
+    fs::path configs_dir;
+    fs::path schemas_dir;
+    fs::path modules_dir;
+    fs::path interfaces_dir;
+    fs::path logging_config;
+    fs::path config_file;
+    bool validate_schema;
+
+    explicit RuntimeSettings(const po::variables_map& vm);
+};
+
+struct ModuleCallbacks {
+    std::function<void(ModuleAdapter::CallFunc call)> register_call_handler;
+    std::function<void(ModuleAdapter::PublishFunc publish)> register_publish_handler;
+    std::function<void(ModuleAdapter::SubscribeFunc subscribe)> register_subscribe_handler;
+    std::function<void(ModuleAdapter::ExtMqttPublishFunc ext_mqtt_publish)> register_ext_mqtt_publish_handler;
+    std::function<void(ModuleAdapter::ExtMqttSubscribeFunc ext_mqtt_subscribe)> register_ext_mqtt_subscribe_handler;
+    std::function<std::vector<cmd>(const json& connections)> everest_register;
+    std::function<void(ModuleConfigs module_configs, const ModuleInfo& info)> init;
+    std::function<void()> ready;
```

Comment:
Probably this commit would be a good point to remove all those different register_handler functions.  Except for the ext_mqtt functions we want to register always all the functions and not just one by one.  I would propose, that instead there will be only one single hook, which the module as a library (in case we'll keep it as a library) will offer - something like register_everest, and this single function will then receive a struct with all the necessary functions to access everest.  So instead of registering n single callbacks, one register with a struct of n callbacks.

Reply:
Sure, that sounds good. I'll do a little more refactoring in this series of PRs.